src/pages/ReportParticipantsPage.tsx

```diff
@@ -1,135 +1,391 @@
-const getAllParticipants = (
-    report: OnyxEntry<Report>,
-    personalDetails: OnyxEntry<PersonalDetailsList>,
-    translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: Localize.PhraseParameters<Localize.Phrase<TKey>>) => string,
-): OptionData[] =>
-    ReportUtils.getVisibleMemberIDs(report)
-        .map((accountID, index) => {
-            const userPersonalDetail = personalDetails?.[accountID];
-            const userLogin =
-                !!userPersonalDetail?.login && !CONST.RESTRICTED_ACCOUNT_IDS.includes(accountID) ? LocalePhoneNumber.formatPhoneNumber(userPersonalDetail.login) : translate('common.hidden');
-            const displayName = PersonalDetailsUtils.getDisplayNameOrDefault(userPersonalDetail);
-
-            const pendingChatMember = report?.pendingChatMembers?.findLast((member) => member.accountID === accountID.toString());
-            return {
-                alternateText: userLogin,
-                pendingAction: pendingChatMember?.pendingAction,
-                displayName,
-                accountID: userPersonalDetail?.accountID ?? accountID,
+type MemberOption = Omit<ListItem, 'accountID'> & {accountID: number};
+
+function ReportParticipantsPage({report, personalDetails, session}: ReportParticipantsPageProps) {
+    const [selectedMembers, setSelectedMembers] = useState<number[]>([]);
+    const [errors, setErrors] = useState({});
+    const [removeMembersConfirmModalVisible, setRemoveMembersConfirmModalVisible] = useState(false);
+    const {translate, formatPhoneNumber} = useLocalize();
+    const styles = useThemeStyles();
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const StyleUtils = useStyleUtils();
+    const selectionListRef = useRef<SelectionListHandle>(null);
+    const textInputRef = useRef<TextInput>(null);
+    const dropdownButtonRef = useRef(null);
+    const getUsers = useCallback((): MemberOption[] => {
+        let result: MemberOption[] = [];
+        ReportUtils.getVisibleMemberIDs(report).forEach((accountID) => {
+            const role = report.participants?.[accountID].role;
+            const details = personalDetails?.[accountID];
+
+            if (!details) {
+                Log.hmmm(`[WorkspaceMembersPage] no personal details found for policy member with accountID: ${accountID}`);
+                return;
+            }
+
+            const isSelected = selectedMembers.includes(accountID);
+
+            const isAdmin = role === CONST.REPORT.ROLE.ADMIN;
+
+            let roleBadge = null;
+            if (isAdmin) {
+                roleBadge = (
+                    <Badge
+                        text={translate('common.admin')}
+                        textStyles={styles.textStrong}
+                        badgeStyles={[styles.justifyContentCenter, StyleUtils.getMinimumWidth(60), styles.badgeBordered, isSelected && styles.activeItemBadge]}
+                    />
+                );
+            }
+
+            result.push({
```

Comment:
Coming from https://github.com/Expensify/App/issues/40442. We missed adding the `isDisabled` param here for members pending deletion.